### PR TITLE
SAK-25018 Discussions allow read-only view of discussions after close date

### PR DESCRIPTION
**Pull request by davidpbauer**
This adds an additional option at the forum and topic levels that allows
users to have a read-only view of discussions after the close date has
passed.

### CODE REVIEW

File: msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/DiscussionForumTool.java [hunk 1]
```diff
@@ -7808,6 +7808,7 @@ private DiscussionTopicBean duplicateTopic(Long originalTopicId, DiscussionForum
 		newTopic.setAvailabilityRestricted(true);
 		newTopic.setOpenDate(fromTopic.getOpenDate());
 		newTopic.setCloseDate(fromTopic.getCloseDate());
+        newTopic.setLockedAfterClosed(fromTopic.getLockedAfterClosed());
```
Comment (ottenhoff) [inline msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/DiscussionForumTool.java:7811]: tab

File: msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/ui/DiscussionTopicBean.java [hunk 2]
```diff
@@ -410,6 +416,20 @@ public void setTopicLocked(Boolean locked)
...
     topic.setLocked(locked);
   }
+
+  private void handleLockedAfterClosedCondition(){
+    Boolean availabilityRestricted = getTopic().getAvailabilityRestricted();
+
+    if(availabilityRestricted && locked.equals(Boolean.FALSE.toString())) {
+      Date closeDate = getTopic().getCloseDate();
+      if (closeDate != null) {
+        if (getTopic().getLockedAfterClosed() && closeDate.before(new Date())) {
```
Comment (ottenhoff) [inline msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/ui/DiscussionTopicBean.java:426]: better to combine these two lines 425/426 into one if

**Review by ottenhoff — COMMENTED** [review]
Looks pretty good!

**Review by adrianfish — APPROVED** [review]
Thanks for the work, Dave.

**Comment by ern** [comment]
needs a rebase

**Comment by davidpbauer** [comment]
> needs a rebase

Rebased on master

**Comment by davidpbauer** [comment]
@ern I've rebased this again and fixed the conflicts.

**Pull request merged**